src/nodes/flow/coupling_flows/planar_flow.jl

```diff
@@ -0,0 +1,258 @@
+    # function when the input is an array with 1 element
+    @assert length(input) == 1 "Something is wrong with the dimensionality of the input to the PlanarFlow flow."
+    return forward(f, input[1])
+end
+forward(f::PlanarFlow{T1,T2}, input::T3) where { T1 <: Real, T2 <: Real, T3 } = _forward(f, input)
+Broadcast.broadcasted(::typeof(forward), f::PlanarFlow{T1,T2}, input::Array{T3,1}) where { T1 <: Real, T2 <: Real, T3 } = broadcast(_forward, Ref(f), input)
+
+# inplace forward pass through the PlanarFlow function (multivariate input)
+function forward!(output::T1, f::PlanarFlow{T1,T2}, input::T1) where { T1, T2 <: Real }
+
+    # check dimensionality
+    @assert length(output) == length(input) "The length of the preallocated vector does not seem to match the length of the input vector."
+
+    # fetch values
+    u, w, b = getall(f)
+    
+    # calculate result
+    output .= u
+    output .*= tanh(dot(w, input) + b) 
+    output .+= input
+
+end
+
+# jacobian of the PlanarFlow function (multivariate input)
+function _jacobian(f::PlanarFlow{T1,T2}, input::T1) where { T1, T2 <: Real}
+
+    # fetch values 
+    u, w, b = getall(f)
+
+    # calculate result
+    result = u*w'
+    result .*= dtanh(dot(w, input) + b)
+    @inbounds for k = 1:length(input)
+        result[k,k] += 1.0
+    end
+
+    # return result
+    return result
+
+end
+jacobian(f::PlanarFlow{T1,T2}, input::T1) where { T1, T2 <: Real } = _jacobian(f, input)
+Broadcast.broadcasted(::typeof(jacobian), f::PlanarFlow{T1,T2}, input::Array{T1,1}) where { T1, T2 <: Real } = broadcast(_jacobian, Ref(f), input)
+
+# jacobian of the PlanarFlow function (univariate input)
+function _jacobian(f::PlanarFlow{T1,T2}, input::T3) where { T1 <: Real, T2 <: Real, T3 <: Real } 
+
+    # fetch values 
+    u, w, b = getall(f)
+
+    # calculate result (optimized)
+    result = u * w * dtanh(w * input + b) + 1
+
+    # return result
+    return result
+
+end
+jacobian(f::PlanarFlow{T1,T2}, input::T3) where { T1 <: Real, T2 <: Real, T3 <: Real } = _jacobian(f, input)
+Broadcast.broadcasted(::typeof(jacobian), f::PlanarFlow{T1,T2}, input::Array{T3,1}) where { T1 <: Real, T2 <: Real, T3 <: Real } = broadcast(_jacobian, Ref(f), input)
+
+function _jacobian(f::PlanarFlow{T1,T2}, input::T3) where { T1 <: Real, T2 <: Real, T3 }
+    # function when the input is an array with 1 element
+    @assert length(input) == 1 "Something is wrong with the dimensionality of the input to the PlanarFlow flow."
+    return jacobian(f, input[1])
+end
+jacobian(f::PlanarFlow{T1,T2}, input::T3) where { T1 <: Real, T2 <: Real, T3 } = _jacobian(f, input)
+Broadcast.broadcasted(::typeof(jacobian), f::PlanarFlow{T1,T2}, input::Array{T3,1}) where { T1 <: Real, T2 <: Real, T3 } = broadcast(_jacobian, Ref(f), input)
+
+# inplace jacobian of the PlanarFlow function (multivariate input)
+function jacobian!(output::Array{T2,2}, f::PlanarFlow{T1,T2}, input::T1) where { T1, T2 <: Real}
```

Comment:
Same as previous comment, better to replace all concrete types from Base to their abstract equivalents, such as `AbstractArray{T2, 2}` or `AbstractMatrix{T2}` (it is an alias for `AbstractArray{T2 , 2}`).

Reply:
Resolved by 0056d3a.